src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -172,6 +184,10 @@ def build_nic_resource(_, name, location, tags, vm_name, subnet_id, private_ip_a
             'count': count
         }
 
+    if edge_zone:
+        nic['extendedLocation'] = edge_zone
+        nic['apiVersion'] = '2021-02-01'
```

Comment:
hard-coded api version?

Reply:
should we get specified api version from resource type and check if the api version is supported?

Reply:
Good question! I've thought about this, and my personal opinion is:

Before, the api-versions of dependent resources on are fixed and independent, so it might be better to Keep the api-versions which support edge zone independent.
Otherwise, there will be several problems:
1. If the api-version of dependent resources is upgraded, the VM related tests need to be recorded, which increases the coupling.
2. The api-version used by dependent resources may tend to be stable and fixed. I think this is the reason why the previous api-version used by dependent resources is the lowest version. [code link](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/vm/_template_builder.py#L72)
3. If other profiles also need to support edge zone in the future, if the api-versions of their dependent resources have not been upgraded in that profile, but we need to give higher priority to support VM and VMSS, then we can only bump version for all the dependent resources together. This workload is very heavy and it is not conducive to split.

In addition, similar situations have been dealt with in this way before: [code link](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/vm/_template_builder.py#L147-L155)

As for why not add the check for api-version, it is because `--edge-zone` has the api-version control `min_api='2020-12-01'` [code link](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/vm/_params.py#L103)

That’s just my two cents, do you have any ideas or suggestions for this?

Reply:
LGTM then